### PR TITLE
posix: caching allocated space in file

### PR DESCRIPTION
**Pull request by igchor**
This change is [Reviewable](https://reviewable.io/reviews/pmem/pmemfile/278)

### CODE REVIEW

File: src/libpmemfile-posix/truncate.c [hunk 1]
```diff
@@ -89,6 +93,11 @@ vinode_truncate(PMEMfilepool *pfp, struct pmemfile_vinode *vinode,
 			TX_SET_DIRECT(inode, mtime, tm);
 			TX_SET_DIRECT(inode, ctime, tm);
 		}
+
+		if (inode->allocated_space != allocated_space) {
+			TX_ADD_DIRECT(&inode->allocated_space);
```
Comment (GBuella) [inline src/libpmemfile-posix/truncate.c]: I guess these TX_ADD_DIRECT + set pairs could be changes to TX_SET_DIRECT instances.

Comment (marcinslusarz) [inline src/libpmemfile-posix/truncate.c]: I'm not a huge fan of TX_SET* macros. IMO they hide too much. But I'm not going to block such changes.

**Comment by marcinslusarz** [comment]
:lgtm:

---

Reviewed 8 of 8 files at r1, 9 of 9 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/278#-:-Kq7O-GwXKF-XJjgWnVX:bnfp4nl)*

**Comment by GBuella** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/278#-:-Kq8FAeuLYxZHsYduWEg:bnfp4nl)*

**Pull request merged**